beacon_node/eth2_libp2p/src/peer_manager/mod.rs

```diff
@@ -491,36 +544,22 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
         true
     }
 
-    /// Notifies the peer manager that this peer is being dialed.
-    pub fn _dialing_peer(&mut self, peer_id: &PeerId) {
-        self.network_globals.peers.write().dialing_peer(peer_id);
-    }
-
-    /// Updates the reputation of known peers according to their connection
+    /// Updates the scores of known peers according to their connection
     /// status and the time that has passed.
-    ///
-    /// **Disconnected peers** get a 1rep hit every hour they stay disconnected.
-    /// **Banned peers** get a 1rep gain for every hour to slowly allow them back again.
-    ///
-    /// A banned(disconnected) peer that gets its rep above(below) MIN_REP_BEFORE_BAN is
-    /// now considered a disconnected(banned) peer.
-    // TODO: Implement when reputation is added.
-    fn _update_reputations(&mut self) {
-        /*
-        // avoid locking the peerdb too often
-        // TODO: call this on a timer
-
-        let now = Instant::now();
-
-        // Check for peers that get banned, unbanned and that should be disconnected
-        let mut ban_queue = Vec::new();
-        let mut unban_queue = Vec::new();
-
+    /// NOTE: This is experimental and will likely be adjusted
+    fn update_peer_scores(&mut self) {
         /* Check how long have peers been in this state and update their reputations if needed */
         let mut pdb = self.network_globals.peers.write();
 
-        for (id, info) in pdb._peers_mut() {
-            // Update reputations
+        let mut to_ban_peers = Vec::new();
+        let mut to_unban_peers = Vec::new();
+
+        for (peer_id, info) in pdb.peers_mut() {
+            let previous_state = info.score.state();
+            // Update scores
+            info.score.update();
+
+            /* TODO: Implement logic about connection lifetimes
```

Comment:
Yeah. This uses a very naive scoring, because I didn't want to break the world with complicated scoring systems. Small steps, debugging then we can add more complex reputations. 
This TODO is about increasing scores based on how long the peer has been connected and its behaviour in that time. 

It's left for a future PR